### PR TITLE
Build separate libvirt client for mocking

### PR DESCRIPTION
**Pull request by ingvagabund**
Must before we can build mock the client and test the libvirt actuator operations in unit tests.

### CODE REVIEW

File: pkg/cloud/libvirt/client/client.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	providerconfigv1 "github.com/openshift/cluster-api-provider-libvirt/pkg/apis/libvirtproviderconfig/v1alpha1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// CreateDomainInput specifies input parameters for CreateDomain operation
+type CreateDomainInput struct {
+	// DomainName is name of domain to be created
+	DomainName string
+
+	// IgnKey is name of existing volume with ignition config (DEPRICATED)
```
Comment (frobware) [inline pkg/cloud/libvirt/client/client.go]: typo: Deprecated.

File: pkg/cloud/libvirt/client/client.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+	// Ignition configuration to be injected during bootstrapping
+	Ignition *providerconfigv1.Ignition
+
+	// CloudInit configuration to be run during bootstrapping
+	CloudInit *providerconfigv1.CloudInit
+
+	// VolumeName of volume to be added to domain definition
+	VolumeName string
+	// PoolName of pool where VolumeName volume is located
```
Comment (frobware) [inline pkg/cloud/libvirt/client/client.go]: nit: consistently add a blank line between fields. Or not. Prefer the former as it easier to read and you already have many fields in this form.

File: pkg/cloud/libvirt/client/client.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	VolumePoolName string
+
+	// NetworkInterfaceName as name of network interface
+	NetworkInterfaceName string
+	// NetworkInterfaceAddress as address of network interface
+	NetworkInterfaceAddress string
+	// HostName as network interface hostname
+	HostName string
+	// Offset as IP address offset
```
Comment (frobware) [inline pkg/cloud/libvirt/client/client.go]: Not sure I understand this field. Offset from what?

File: pkg/cloud/libvirt/client/client.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	Offset int
+
+	// Autostart as domain autostart
+	Autostart bool
+
+	// Memory allocated for running domain
```
Comment (frobware) [inline pkg/cloud/libvirt/client/client.go]: Comment doesn't match field name.

File: pkg/cloud/libvirt/client/client.go [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+	DomainMemory int
+	// Vcpu allocated for running domain
```
Comment (frobware) [inline pkg/cloud/libvirt/client/client.go]: Comment doesn't match field name.

File: pkg/cloud/libvirt/client/client.go [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+	CreateDomain(*CreateDomainInput) error
+	// DeleteDomain deletes a domain
+	DeleteDomain(name string) error
+	// DomainExists checks if domain exists
+	DomainExists(name string) (bool, error)
+	// LookupDomainByName looks up a domain based on its name
+	LookupDomainByName(name string) (*libvirt.Domain, error)
+
+	// CreateVolume creates volume based on CreateVolumeInput
+	CreateVolume(*CreateVolumeInput) error
```
Comment (frobware) [inline pkg/cloud/libvirt/client/client.go]: Need this be a pointer? what happens if I pass `nil`?

File: pkg/cloud/libvirt/client/client.go [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+	VolumeFormat string
+}
+
+// Client is a wrapper object for actual libvirt library to allow for easier testing.
+type Client interface {
+	// Close closes the client's libvirt connection.
+	Close() error
+
+	// CreateDomain creates domain based on CreateDomainInput
+	CreateDomain(*CreateDomainInput) error
```
Comment (frobware) [inline pkg/cloud/libvirt/client/client.go]: Need this be a pointer? What happens if I pass `nil`?

**Comment by ingvagabund** [comment]
Removing approve until the CI is fixed.

**Comment by ingvagabund** [comment]
Fixing CI through https://github.com/openshift/aos-cd-jobs/pull/1677

**Pull request merged**